### PR TITLE
solidstart-feat: modified eslint rules adding typescript plugin

### PR DESCRIPTION
**Pull request by Megio**
Added typescript plugin to eslint and configured it.

Fixed: #1409

### CODE REVIEW

**Review by SDaian — DISMISSED** [review]
LGTM 🎸

File: qwik-graphql-tailwind/.storybook/preview.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 // Based on https://github.com/BuilderIO/qwik/discussions/787#discussioncomment-3715103
-import { JSXNode, useContextProvider } from '@builder.io/qwik';
```
Comment (hdJerry) [inline qwik-graphql-tailwind/.storybook/preview.ts:2]: this file shouldn't be touched, can you revert this one

**Review by hdJerry — CHANGES_REQUESTED** [review]

File: qwik-graphql-tailwind/.storybook/preview.ts (hunk 1 above)
Comment (Megio) [reply qwik-graphql-tailwind/.storybook/preview.ts:2]: that import was unused, do we still need that `useContextProvider`?

**Review by hdJerry — APPROVED** [review]
Thanks!!!

**Pull request merged**